source/initial_conditions/adiabatic.cc

```diff
@@ -58,6 +52,21 @@ namespace aspect
                                                   :
                                                   adiabatic_surface_temperature;
 
+      // then, get the temperature at the top and bottom boundary of the model
+      // if no boundary temperature is prescribed simply use the adiabatic
+      const double T_surface = (&this->get_boundary_temperature() != 0)
+                               ?
+                                   this->get_boundary_temperature().minimal_temperature(
+                                       this->get_fixed_temperature_boundary_indicators())
+                               :
+                                   adiabatic_surface_temperature;
+      const double T_bottom = (&this->get_boundary_temperature() != 0)
+                              ?
+                                  this->get_boundary_temperature().maximal_temperature(
+                                      this->get_fixed_temperature_boundary_indicators())
+                              :
+                                  adiabatic_surface_temperature;
+
```

Comment:
This assumes that a) surface and bottom are Dirichlet boundaries and b) that the min is at the surface and the max is at the bottom, right? What happens if I only set the temperature on the side walls? Should we update the comment above this code?